auacm/app/modules/competition_manager/views.py

```diff
@@ -197,57 +202,90 @@ def get_competition_data(cid):
     })
 
 
+@app.route('/api/competitions/<int:cid>', methods=['DELETE'])
+def delete_competition(cid):
+    """Delete a competition and all the data associated with it"""
+    competition = (database.session.query(Competition)
+                   .filter_by(cid=cid).first())
+
+    if not competition:
+        return '404: Competition not found', 404
+
+    # Delete the problems from a competition
+    competition_problems = (database.session.query(CompProblem)
+                            .filter_by(cid=cid).all())
+    for competition_problem in competition_problems:
+        database.session.delete(competition_problem)
+
+    # Delete the users/tams from a competition
+    competition_users = (database.session.query(CompUser)
+                         .filter_by(cid=cid).all())
+    for user in competition_users:
+        database.session.delete(user)
+
+    # Delete the competition itself
+    database.session.delete(competition)
+    database.session.commit()
+
+    return 'Successfully deleted', 204
```

Comment:
After more thought, I think 204 is actually good. I do think that we should modify it so that we use the `serve_respose()` function though rather than simply returning a string if nothing more just to conform to the rest of the codebase. It might take a couple lines of modification to allow different response codes, but it shouldn't be bad.